core/src/main/scala/org/apache/spark/api/java/JavaRDDLike.scala

```diff
@@ -214,8 +214,8 @@ trait JavaRDDLike[T, This <: JavaRDDLike[T, This]] extends Serializable {
   /**
    * Applies a function f to each partition of this RDD.
    */
-  def foreachPartition(f: VoidFunction[java.util.Iterator[T]]) {
-    rdd.foreachPartition(x => f.call(x.asJava))
+  def foreachPartition(f: VoidFunction[JIterator[T]]) {
+    rdd.foreachPartition((x => f.call(x.asJava)))
```

Comment:
Nit: this added extra parens which aren't needed. If you wouldn't mind, while changing this, end the line above with `): Unit = {` to standardize